src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs

```diff
@@ -456,5 +456,37 @@ internal override bool IsAwaitKeywordContext()
 
             return false;
         }
+
+        internal override bool IsDotAwaitKeywordContext(CancellationToken cancellationToken)
```

Comment:
i feel like this could be pulled into AbstractAwaitCompletionProvider.cs directly, or have it be abstract there and implemented in subclasses. t his is a rather odd feature-specific extension to have right here.